### ISSUE TITLE
Opening preferences dialog crashes the app

### ISSUE THREAD

**Issue by dimkab** [xournalpp/xournalpp#2591]
- OS: MacOS 11.1 Big Sur
- libgtk 3.24.10
- Xournal++ 1.0.20
- Installed from the official release

Clicking Edit->Preferences crashes the app.

Note: this does not happen in 1.0.18/1.0.19 running on the same OS.

EDIT: the console output and the crash log are below.
```
/Applications Xournal++\ 1.0.20.app/Contents/MacOS/xournalpp
2020-12-28 01:24:49.829 defaults[33059:864416] 
The domain/default pair of (.GlobalPreferences, AppleCollationOrder) does not exist

** (com.github.xournalpp.xournalpp:33053): WARNING **: 01:24:50.656: No device found. Is Xournal++ running in debugger / Eclipse...?
Probably this is the reason for not finding devices!


(com.github.xournalpp.xournalpp:33053): GLib-GObject-WARNING **: 01:24:50.688: g_object_get_is_valid_property: object class 'GtkLabel' has no property named 'accel-closure'

(com.github.xournalpp.xournalpp:33053): Gtk-WARNING **: 01:24:50.961: Drawing a gadget with negative dimensions. Did you forget to allocate a size? (node marks owner GtkScale)

(com.github.xournalpp.xournalpp:33053): Gtk-WARNING **: 01:24:50.961: Drawing a gadget with negative dimensions. Did you forget to allocate a size? (node mark owner GtkScale)

(com.github.xournalpp.xournalpp:33053): Gtk-WARNING **: 01:24:50.961: Drawing a gadget with negative dimensions. Did you forget to allocate a size? (node indicator owner GtkScale)

(com.github.xournalpp.xournalpp:33053): Gtk-WARNING **: 01:24:50.961: Drawing a gadget with negative dimensions. Did you forget to allocate a size? (node mark owner GtkScale)

(com.github.xournalpp.xournalpp:33053): Gtk-WARNING **: 01:24:50.961: Drawing a gadget with negative dimensions. Did you forget to allocate a size? (node indicator owner GtkScale)

** (com.github.xournalpp.xournalpp:33053): WARNING **: 01:25:04.851: No device found. Is Xournal++ running in debugger / Eclipse...?
Probably this is the reason for not finding devices!


** (com.github.xournalpp.xournalpp:33053): WARNING **: 01:25:04.870: No device found. Is Xournal++ running in debugger / Eclipse...?
Probably this is the reason for not finding devices!


** (com.github.xournalpp.xournalpp:33053): WARNING **: 01:25:05.250: [Crash Handler] Crashed with signal 11

** (com.github.xournalpp.xournalpp:33053): WARNING **: 01:25:05.250: [Crash Handler] Wrote crash log to: /Users/dima/.xournalpp/errorlogs/errorlog.20201228-012505.log
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found
sh: addr2line: command not found

** (com.github.xournalpp.xournalpp:33053): WARNING **: 01:25:05.339: Trying to emergency save the current open document…

** (com.github.xournalpp.xournalpp:33053): WARNING **: 01:25:05.340: Successfully saved document to "/Users/dima/.xournalpp//emergencysave.xopp"

```

The crash log:
```
cat /Users/dima/.xournalpp/errorlogs/errorlog.20201228-012505.log
Date: Mon Dec 28 01:25:05 2020
Error: signal 11
[bt]: (0) 0   xournalpp-bin                       0x000000010f3cf974 _ZL12crashHandleri + 84
[bt]: (1) 1   libsystem_platform.dylib            0x00007fff20395d7d _sigtramp + 29
[bt]: (2) 2   ???                                 0x00007ffee0987980 0x0 + 140732666509696
[bt]: (3) 3   libcairo.2.dylib                    0x000000011054f138 DataProviderReleaseCallback + 56
[bt]: (4) 4   CoreGraphics                        0x00007fff24fbe8cd data_release_info + 31
[bt]: (5) 5   CoreGraphics                        0x00007fff24f6fda1 data_provider_finalize + 64
[bt]: (6) 6   CoreGraphics                        0x00007fff24f35dad data_provider_retain_count + 74
[bt]: (7) 7   CoreFoundation                      0x00007fff20513dd8 _CFRelease + 474
[bt]: (8) 8   CoreGraphics                        0x00007fff24f9bb3f image_finalize + 103
[bt]: (9) 9   CoreFoundation                      0x00007fff20513cf2 _CFRelease + 244
[bt]: (10) 10  CoreGraphics                        0x00007fff24fe7bbf _ZN2CG24DisplayListResourceImageD2Ev + 23
[bt]: (11) 11  CoreGraphics                        0x00007fff24fe7b92 _ZN2CG24DisplayListResourceImageD0Ev + 14
[bt]: (12) 12  CoreGraphics                        0x00007fff24fe7b5f _ZNSt3__110shared_ptrIN2CG24DisplayListResourceImageEED2Ev + 49
[bt]: (13) 13  CoreGraphics                        0x00007fff2536d577 _ZNSt3__16__treeINS_10shared_ptrIN2CG24DisplayListResourceImageEEENS2_20CompareResourceImageENS_9allocatorIS4_EEE7destroyEPNS_11__tree_nodeIS4_PvEE + 47
[bt]: (14) 14  CoreGraphics                        0x00007fff2536d56e _ZNSt3__16__treeINS_10shared_ptrIN2CG24DisplayListResourceImageEEENS2_20CompareResourceImageENS_9allocatorIS4_EEE7destroyEPNS_11__tree_nodeIS4_PvEE + 38
[bt]: (15) 15  CoreGraphics                        0x00007fff2536d56e _ZNSt3__16__treeINS_10shared_ptrIN2CG24DisplayListResourceImageEEENS2_20CompareResourceImageENS_9allocatorIS4_EEE7destroyEPNS_11__tree_nodeIS4_PvEE + 38
[bt]: (16) 16  CoreGraphics                        0x00007fff24fe7754 _ZN2CG11DisplayListD2Ev + 316
[bt]: (17) 17  CoreGraphics                        0x00007fff24fe75fd destroyDisplayList2Impl + 26
[bt]: (18) 18  CoreGraphics                        0x00007fff24fe75cf dl2_Finalize + 23
[bt]: (19) 19  CoreFoundation                      0x00007fff20513cf2 _CFRelease + 244
[bt]: (20) 20  QuartzCore                          0x00007fff26c043e0 CABackingStoreUpdate_ + 925
[bt]: (21) 21  QuartzCore                          0x00007fff26c69279 ___ZN2CA5Layer8display_Ev_block_invoke + 53
[bt]: (22) 22  QuartzCore                          0x00007fff26c0366c -[CALayer _display] + 2183
[bt]: (23) 23  AppKit                              0x00007fff22dd4001 -[_NSBackingLayer display] + 475
[bt]: (24) 24  AppKit                              0x00007fff22d3f535 -[_NSViewBackingLayer display] + 555
[bt]: (25) 25  QuartzCore                          0x00007fff26c02660 _ZN2CA5Layer17display_if_neededEPNS_11TransactionE + 874
[bt]: (26) 26  QuartzCore                          0x00007fff26d4273b _ZN2CA7Context18commit_transactionEPNS_11TransactionEdPd + 517
[bt]: (27) 27  QuartzCore                          0x00007fff26be3fd5 _ZN2CA11Transaction6commitEv + 783
[bt]: (28) 28  AppKit                              0x00007fff22de9272 __62+[CATransaction(NSCATransaction) NS_setFlushesWithDisplayLink]_block_invoke + 285
[bt]: (29) 29  AppKit                              0x00007fff23538085 ___NSRunLoopObserverCreateWithHandler_block_invoke + 41
[bt]: (30) 30  CoreFoundation                      0x00007fff20448dad __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 23
[bt]: (31) 31  CoreFoundation                      0x00007fff20448c3d __CFRunLoopDoObservers + 549
[bt]: (32) 32  CoreFoundation                      0x00007fff204480ed __CFRunLoopRun + 838
[bt]: (33) 33  CoreFoundation                      0x00007fff204476ce CFRunLoopRunSpecific + 563
[bt]: (34) 34  HIToolbox                           0x00007fff286cc6d0 RunCurrentEventLoopInMode + 292
[bt]: (35) 35  HIToolbox                           0x00007fff286cc322 ReceiveNextEventCommon + 283
[bt]: (36) 36  HIToolbox                           0x00007fff286cc1ef _BlockUntilNextEventMatchingListInModeWithFilter + 64
[bt]: (37) 37  AppKit                              0x00007fff22c64de9 _DPSNextEvent + 883
[bt]: (38) 38  AppKit                              0x00007fff22c635af -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1366
[bt]: (39) 39  libgdk-3.0.dylib                    0x000000011008c276 poll_func + 278
[bt]: (40) 40  libglib-2.0.0.dylib                 0x000000010f659f06 g_main_context_iterate + 422
[bt]: (41) 41  libglib-2.0.0.dylib                 0x000000010f65a27f g_main_loop_run + 191
[bt]: (42) 42  libgtk-3.0.dylib                    0x000000010f9511d3 gtk_dialog_run + 579
[bt]: (43) 43  xournalpp-bin                       0x000000010f34a88d _ZN14SettingsDialog4showEP10_GtkWindow + 205
[bt]: (44) 44  xournalpp-bin                       0x000000010f28bb7c _ZN7Control12showSettingsEv + 332
[bt]: (45) 45  xournalpp-bin                       0x000000010f373944 _ZN12AbstractItem9activatedEP9_GdkEventP12_GtkMenuItemP14_GtkToolButton + 532
[bt]: (46) 46  libgobject-2.0.0.dylib              0x000000011087464c g_closure_invoke + 204
[bt]: (47) 47  libgobject-2.0.0.dylib              0x000000011088a8f0 signal_emit_unlocked_R + 1744
[bt]: (48) 48  libgobject-2.0.0.dylib              0x000000011088b604 g_signal_emit_valist + 2100
[bt]: (49) 49  libgobject-2.0.0.dylib              0x000000011088bd82 g_signal_emit + 130
[bt]: (50) 50  libgtk-3.0.dylib                    0x000000010fa58ee6 gtk_menu_item_activate + 198
[bt]: (51) 51  libgobject-2.0.0.dylib              0x000000011087464c g_closure_invoke + 204
[bt]: (52) 52  libgtkmacintegration-gtk3.2.dylib   0x0000000110bb4c71 idle_call_activate + 113
[bt]: (53) 53  libgdk-3.0.dylib                    0x0000000110035827 gdk_threads_dispatch + 71
[bt]: (54) 54  libglib-2.0.0.dylib                 0x000000010f659bd6 g_main_context_dispatch + 310
[bt]: (55) 55  libglib-2.0.0.dylib                 0x000000010f659f62 g_main_context_iterate + 514
[bt]: (56) 56  libglib-2.0.0.dylib                 0x000000010f65a27f g_main_loop_run + 191
[bt]: (57) 57  libgtk-3.0.dylib                    0x000000010fa4079c gtk_main + 108
[bt]: (58) 58  xournalpp-bin                       0x000000010f2ac38e _ZN11XournalMain3runEiPPc + 3614
[bt]: (59) 59  xournalpp-bin                       0x000000010f3cf878 main + 56
[bt]: (60) 60  libdyld.dylib                       0x00007fff2036c621 start + 1
[bt]: (61) 61  ???                                 0x0000000000000001 0x0 + 1


Try to get a better stracktrace...
[bt] #1 1   xournalpp-bin                       0x000000010f3cffda _ZL12crashHandleri + 1722
[bt] #2 2   libsystem_platform.dylib            0x00007fff20395d7d _sigtramp + 29
[bt] #3 3   ???                                 0x00007ffee0987980 0x0 + 140732666509696
[bt] #4 4   libcairo.2.dylib                    0x000000011054f138 DataProviderReleaseCallback + 56
[bt] #5 5   CoreGraphics                        0x00007fff24fbe8cd data_release_info + 31
[bt] #6 6   CoreGraphics                        0x00007fff24f6fda1 data_provider_finalize + 64
[bt] #7 7   CoreGraphics                        0x00007fff24f35dad data_provider_retain_count + 74
[bt] #8 8   CoreFoundation                      0x00007fff20513dd8 _CFRelease + 474
[bt] #9 9   CoreGraphics                        0x00007fff24f9bb3f image_finalize + 103
[bt] #10 10  CoreFoundation                      0x00007fff20513cf2 _CFRelease + 244
[bt] #11 11  CoreGraphics                        0x00007fff24fe7bbf _ZN2CG24DisplayListResourceImageD2Ev + 23
[bt] #12 12  CoreGraphics                        0x00007fff24fe7b92 _ZN2CG24DisplayListResourceImageD0Ev + 14
[bt] #13 13  CoreGraphics                        0x00007fff24fe7b5f _ZNSt3__110shared_ptrIN2CG24DisplayListResourceImageEED2Ev + 49
[bt] #14 14  CoreGraphics                        0x00007fff2536d577 _ZNSt3__16__treeINS_10shared_ptrIN2CG24DisplayListResourceImageEEENS2_20CompareResourceImageENS_9allocatorIS4_EEE7destroyEPNS_11__tree_nodeIS4_PvEE + 47
[bt] #15 15  CoreGraphics                        0x00007fff2536d56e _ZNSt3__16__treeINS_10shared_ptrIN2CG24DisplayListResourceImageEEENS2_20CompareResourceImageENS_9allocatorIS4_EEE7destroyEPNS_11__tree_nodeIS4_PvEE + 38
[bt] #16 16  CoreGraphics                        0x00007fff2536d56e _ZNSt3__16__treeINS_10shared_ptrIN2CG24DisplayListResourceImageEEENS2_20CompareResourceImageENS_9allocatorIS4_EEE7destroyEPNS_11__tree_nodeIS4_PvEE + 38
[bt] #17 17  CoreGraphics                        0x00007fff24fe7754 _ZN2CG11DisplayListD2Ev + 316
[bt] #18 18  CoreGraphics                        0x00007fff24fe75fd destroyDisplayList2Impl + 26
[bt] #19 19  CoreGraphics                        0x00007fff24fe75cf dl2_Finalize + 23
[bt] #20 20  CoreFoundation                      0x00007fff20513cf2 _CFRelease + 244
[bt] #21 21  QuartzCore                          0x00007fff26c043e0 CABackingStoreUpdate_ + 925
[bt] #22 22  QuartzCore                          0x00007fff26c69279 ___ZN2CA5Layer8display_Ev_block_invoke + 53
[bt] #23 23  QuartzCore                          0x00007fff26c0366c -[CALayer _display] + 2183
[bt] #24 24  AppKit                              0x00007fff22dd4001 -[_NSBackingLayer display] + 475
[bt] #25 25  AppKit                              0x00007fff22d3f535 -[_NSViewBackingLayer display] + 555
[bt] #26 26  QuartzCore                          0x00007fff26c02660 _ZN2CA5Layer17display_if_neededEPNS_11TransactionE + 874
[bt] #27 27  QuartzCore                          0x00007fff26d4273b _ZN2CA7Context18commit_transactionEPNS_11TransactionEdPd + 517
[bt] #28 28  QuartzCore                          0x00007fff26be3fd5 _ZN2CA11Transaction6commitEv + 783
[bt] #29 29  AppKit                              0x00007fff22de9272 __62+[CATransaction(NSCATransaction) NS_setFlushesWithDisplayLink]_block_invoke + 285
[bt] #30 30  AppKit                              0x00007fff23538085 ___NSRunLoopObserverCreateWithHandler_block_invoke + 41
[bt] #31 31  CoreFoundation                      0x00007fff20448dad __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 23
```

**Comment by rolandlo** [comment]
Unfortunately I don't have MacOS for testing. I experience something similar with the AppImage though. Does the app also crash if you click on menu `Help->About`?

**Comment by dimkab** [comment]
> Unfortunately I don't have MacOS for testing. I experience something similar with the AppImage though. Does the app also crash if you click on menu `Help->About`?

Nope, the About dialog displays just fine.

**Comment by Febbe** [comment]
Can you run xournalpp from the terminal and send us the console output?
There should also be a crash log in your xournalpp configdir(%XDG_CONFIG_DIR%/xournalpp).

**Comment by dimkab** [comment]
> Can you run xournalpp from the terminal and send us the console output?
> There should also be a crash log in your xournalpp configdir(%XDG_CONFIG_DIR%/xournalpp).

Updated the main issue text.

**Comment by rolandlo** [comment]
With the nightly release built tonight it shouldn't crash anymore.

**Comment by rolandlo** [comment]
Closing due to inactivity. It's most likely fixed in version 1.1.0
